pkg/apis/autoscaling/annotation_validation.go

```diff
@@ -145,6 +146,12 @@ func validateMinMaxScale(annotations map[string]string) *apis.FieldError {
 			Paths:   []string{MaxScaleAnnotationKey, MinScaleAnnotationKey},
 		})
 	}
+	if config.MaxScaleLimit > 0 && (max == 0 || max > int64(config.MaxScaleLimit)) {
```

Comment:
I am unsure if we want to prohibit `0` though 🤔

Reply:
does this consider DefaultMaxScale? Ie if DefaultMaxScale is set to 10, is 0 allowed?

Reply:
> does this consider DefaultMaxScale? Ie if DefaultMaxScale is set to 10, is 0 allowed?

DefaultMaxScale is 0. I guess you mean `if MaxScaleLimit is set to 10, is 0 (default value) allowed`?

Reply:
I mean if operator has overridden DefaultMaxScale (possible as of https://github.com/knative/serving/commit/7c8536801855c08947635016934190a212e8fdb1 ) so 0 means 10, does this work properly?

Reply:
All that said, I think we should just permit `0` in general (whether default MaxScale is set or not, but especially if it is).

Reply:
I see. The original issue does ask for 0 not to be allowed: `Setting this would then ban the use of zero (infinite)`. I can remove this part.

Reply:
hm, fwiw I can't think of a case where an operator wants to limit you to MaxScaleLimit of (eg) 10, but is totally fine with you having unlimited instances. I think 0 should be allowed, but only if it there's a non-unlimited default set.

Reply:
(This is actually the same as with ContainerConcurrencyMaxLimit, there we ended up having to add a new `allow-container-concurrency-zero` flag to prevent unlimited to avoid backwards incompatibility in https://github.com/knative/serving/pull/7932, we could do the same thing here, but... I vote we just say you can only set 0 if there's a non-unlimited default and we can add a flag to change that if someone comes up for a use case for it -- I can't think of one honestly, tho)

Reply:
As I said in another comment, it becomes much more palatable if we verify we only do this on create, so if we verify to ignore existing services — 🤷

Reply:
So validate not-unlimited if limit set, but only for new revisions? Yeah, sgtm